### PR TITLE
owners & species

### PR DESCRIPTION
**Pull request by ybadr99**
add 2 tables owner and species and test some joins queries
Screenshot from 2023-07-13 21-29-46
Screenshot from 2023-07-13 21-34-07
Screenshot from 2023-07-13 21-34-44
Screenshot from 2023-07-13 21-35-32
Screenshot from 2023-07-13 21-35-55

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -1,10 +1,21 @@
...
 CREATE TABLE animals (
     id INT PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
     name VARCHAR(255) NOT NULL,
     date_of_birth DATE,
     escape_attempts INT,
     neutered BOOLEAN,
-    weight_kg DECIMAL
+    weight_kg DECIMAL,
+    species_id INT REFERENCES species(id),
+    owner_id INT REFERENCES owners(id)
```
Comment (franclobo) [inline schema.sql]: - [ ] Kindly use ALTER TABLE command to modify the animals' table 👇🏽 

```sql
ALTER TABLE animals DROP column species;
ALTER TABLE animals ADD species_id int REFERENCES species(id);
ALTER TABLE animals ADD owner_id int REFERENCES owners(id);
```

File: queries.sql [hunk 2]
```diff
@@ -51,3 +51,31 @@ WHERE escape_attempts > 0;
...
+-- What animals belong to Melody Pond?
+SELECT * FROM animals INNER JOIN owners ON animals.owner_id = owners.id WHERE owners.full_name='Melody Pond';
+-- List of all animals that are pokemon (their type is Pokemon).
+SELECT animals.* FROM animals
+LEFT JOIN species ON animals.species_id = species.id
+WHERE species.name = 'Pokemon';
+-- List all owners and their animals, remember to include those that don't own any animal.
+SELECT owners.full_name AS owner_name , animals.name AS animal_name
+FROM owners
+INNER JOIN animals ON owners.id = animals.owner_id;
```
Comment (franclobo) [inline queries.sql]: - [ ] `Jodie Whittaker` doesn't own any animal but she doesn't appears on the list, kindly try to join the tables in another way.

image

File: queries.sql [hunk 3]
```diff
@@ -51,3 +51,31 @@ WHERE escape_attempts > 0;
...
+SELECT animals.* FROM animals
+LEFT JOIN species ON animals.species_id = species.id
+WHERE species.name = 'Pokemon';
+-- List all owners and their animals, remember to include those that don't own any animal.
+SELECT owners.full_name AS owner_name , animals.name AS animal_name
+FROM owners
+INNER JOIN animals ON owners.id = animals.owner_id; 
+-- How many animals are there per species?
+SELECT species.name, count(*) FROM animals
+INNER JOIN species ON animals.species_id = species.id GROUP BY species.name;
```
Comment (franclobo) [inline queries.sql]: - [ ] There are 10 animals assigned to the owners, please 🙏🏽 correct this query to return the correct answer.

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @ybadr99, 

# Required Changes ♻️

so-close-jim-halpert

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

- Correct data ✔️ 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by iambenkis — APPROVED** [review]
# Approved 👏  
### **Highlights 🚀 :heavy_check_mark: :heavy_check_mark:** 

- Well designed schema :heavy_check_mark:
- Excellent queries :heavy_check_mark: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### **Congratulations!** 🎉 
image

## Optional Changes ♻️ 

- [ ] [OPTIONAL] Good job adding customizing this section 👍 , however you misplaced it , Kindly put it in the right place as represented in the README template , make sure you don't have it at the top of the table of contents 👍 
Capture

- [ ] [OPTIONAL] Under the MIT license , the note text is found while it contains text unrelated to you project , kindly delete them , they will make your project documentation even more professional 👍 
Capture

- [ ] [OPTIONAL] Why don't you use the correct project name ? , Kindly  use the name that gives an idea of what you did 👍 , like `vet_clinic db` 😄 😄 😄 

Capture



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @iambenkis in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**